### PR TITLE
mask referrer in posthog

### PR DESCRIPTION
**Pull request by jacekkolasa**
- Change several posthog bindings
- remove GTM and enable posthog without feature-flag
- add separate bindings for demo servers

fixes https://github.com/netdata/dashboard/issues/240,
https://github.com/netdata/dashboard/issues/241
https://github.com/netdata/dashboard/issues/244
https://github.com/netdata/dashboard/issues/245
https://github.com/netdata/dashboard/issues/246
https://github.com/netdata/dashboard/issues/248

### CODE REVIEW

File: src/domains/global/sagas.ts [hunk 1]
```diff
@@ -104,14 +105,23 @@ function* injectPosthog(machineGuid: string, personGuid?: string) {
...
       {
         $ip: "127.0.0.1",
-        $current_url: "agent dashboard",
-        $pathname: "netdata-dashboard",
-        $host: "dashboard.netdata.io",
+        $current_url: isDemo ? null : "agent dashboard",
+        $pathname: isDemo ? null : "netdata-dashboard",
+        $host: isDemo ? null : "dashboard.netdata.io",
+
+        $initial_referring_domain: shouldMaskReferrer ? "127.0.0.1" : null,
```
Comment (andrewm4894) [inline src/domains/global/sagas.ts:118]: Can we use "masked" instead of "127.0.0.1"

File: src/domains/global/sagas.ts [hunk 2]
```diff
@@ -104,14 +105,23 @@ function* injectPosthog(machineGuid: string, personGuid?: string) {
...
+        $initial_referrer: shouldMaskReferrer ? "http://127.0.0.1" : null,
```
Comment (andrewm4894) [inline src/domains/global/sagas.ts:119]: Can we use "masked" instead of "http://127.0.0.1"

File: src/domains/global/sagas.ts [hunk 3]
```diff
@@ -104,14 +105,23 @@ function* injectPosthog(machineGuid: string, personGuid?: string) {
...
+        $referring_domain: shouldMaskReferrer ? "127.0.0.1" : null,
```
Comment (andrewm4894) [inline src/domains/global/sagas.ts:120]: Can we use "masked" instead of "127.0.0.1"

File: src/domains/global/sagas.ts [hunk 4]
```diff
@@ -104,14 +105,23 @@ function* injectPosthog(machineGuid: string, personGuid?: string) {
...
+        $referrer: shouldMaskReferrer ? "http://127.0.0.1" : null,
```
Comment (andrewm4894) [inline src/domains/global/sagas.ts:121]: Can we use "masked" instead of "http://127.0.0.1"

File: src/domains/global/utils.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  "",
+  "https://www.google.com/",
+  "https://duckduckgo.com/",
+  "https://www.reddit.com/",
+]
+
+export const isAllowedReferrer = (referrer: string) => allowedReferrerDomains.includes(referrer)
+  || referrer.endsWith(".my-netdata.io/")
+  || referrer.startsWith("https://github.com/")
+  || referrer.endsWith("netdata.cloud/")
```
Comment (andrewm4894) [inline src/domains/global/utils.ts:11]: Should we add a 
```
|| referrer.startsWith("https://app.netdata.cloud/")
```

As i do also see some referrer strings like this:

https://app.netdata.cloud/spaces/andrewm4894/rooms/general/overview

Which could be very useful.

**Review by andrewm4894 — CHANGES_REQUESTED** [review]
Just one or two small changes as i want to be as explicit as possible about data we mask so think using "masked" as opposed to "127.0.0.1" should be clearer for anyone using or looking at the data downstream anywhere.

**Review by novykh — APPROVED** [review]

File: src/domains/global/sagas.ts (hunk 4 above)
Comment (jacekkolasa) [reply src/domains/global/sagas.ts:89]: sure, done in all 4 lines that you mentioned

**Pull request merged**